Katello/katello
# Fixes #33296 - Publish OSTree repos to a content view

### What are the changes introduced in this pull request?

OSTree repositories are publishable in content view versions and most required information is available to view.  The content count for OSTree (and other generic types) is viewable on the main content view version table and the repositories table within a content view version.  This PR does not add the vertical left-side tabs for generic content.


![Screenshot from 2021-11-09 14-42-54](https://user-images.githubusercontent.com/14796566/140993660-f237ef93-b34e-4fd9-9ca6-576cb187f104.png)

![Screenshot from 2021-11-09 14-43-11](https://user-images.githubusercontent.com/14796566/140993676-779abb87-0597-44a8-b234-74c562b468b3.png)

### What is the thinking behind these changes?

- I named the OSTree content "OSTree Refs" because that's what Pulp calls them and it seems to be the current popular name.  https://manpages.debian.org/testing/ostree/ostree-refs.1.en.html
- I created an entry for OSTree in the `ContentConfig` to store the same kinds of hardcoded information that the other generic types have.
- I used the `ContentConfig` to generate the content-count-related information.

### What are the testing steps for this pull request?

1) Sync a Python repo and an OSTree repo and add them both to a content view.
2) Sync the content view and ensure you see the content counts on both the content  view versions table and within each content view version.
3) Check that the links work in the repositories table within the content view versions.
  -> Note: the content counts on the content view versions themselves don't have links yet because the content tabs need to be created for generic repositories
4) Check that `N/A` appears under "Additional content" for content view versions with no content other than RPMs and Errata.
5) Ensure that no other parts of the content view pages have been broken.

TODO:

- [x] Tests.  This PR is a draft but I figured I'd open it up to reviews while I figure the tests out.

Let me know if there are more tests that I should add.
